### DISCUSSION TITLE
objective function including factorial function

### DISCUSSION THREAD

**Discussion by ellensonny** [yalmip/YALMIP#1345] [General discussion] [answered]
dear professor,
I defined a variable“ intvar k”.
and i want to write the objective function including factorial function,like this "factorial(k)".
but something went wrong.
how i can do to solve the problem?
if there any simple ways to write the objective function?

**Comment by johanlofberg** [comment]
You would have to define "something went wrong" better

**Reply by ellensonny** [reply]
thanks for your reply.
But ,sorry i don not understand.You mean that i should write something to replace the "factorial function" indeed?

**Reply by johanlofberg** [reply]
I meant you should explain what kind of problem you have.

I guess you have the error messsage

```
Error using factorial (line 20)
N must be an array of real non-negative integers.
 
```

I though factorial was overloaded on sdpvars, but it is not. Hence you have to implement it manually https://yalmip.github.io/tutorial/logicprogramming#:~:text=y%20%3D%20f(x)%2C%20x%20scalar%20integer

i.e. 

```
range = 0:N
intvar x f
z = binvar(length(range),1)
Model = [x == range*z, f == factorial(range)*z, sum(z)==1]
```

**Comment by ellensonny** [comment]
sorry not to experss my question appropriately.actually my objective function like this：
![Snipaste_2023-11-07_17-19-52](https://github.com/yalmip/YALMIP/assets/130203697/c88f4438-93c5-4f10-b1ef-9ae1c786d2a4)
"k" is the only intvar variable.and the other is constant.
my question is that how can i write the objective function easily.

**Reply by johanlofberg** [reply, marked as answer]
Same thing, you just have a very nasty function

```
range = 0:N
intvar k
sdpvar t
z = binvar(length(range),1)
Model = [k == range*z, t == myhorriblefunction(range)*z, sum(z)==1]
```

**Reply by ellensonny** [reply]
thank you so much!Your response helped me slove the problem effectively.
